src/tests/test_known_external.rs

```diff
@@ -39,6 +39,34 @@ fn known_external_complex_unknown_args() -> TestResult {
     )
 }
 
+#[test]
+fn known_external_from_module() -> TestResult {
+    run_test_contains(
+        r#"module spam {
+            export extern echo []
+        }
+
+        use spam echo
+        echo foo -b -as -9 --abc -- -Dxmy=AKOO - bar
+        "#,
+        "foo -b -as -9 --abc -- -Dxmy=AKOO - bar",
+    )
+}
+
+#[test]
+fn known_external_wrapped_from_module() -> TestResult {
+    run_test_contains(
+        r#"module spam {
+            export extern-wrapped my-echo [...rest] { ^echo $rest }
```

Comment:
Was thinking that we have a `--testbin` for `echo` that works better cross-platform. Seems like we've had problems with `^echo` in the past. Just a suggestion.

Reply:
I tried using the testbin, but couldn't get it to work. `echo` is used in other extern tests and so has worked fine.

Reply:
I'm fine with whatever you think is best. Maybe it works better in the repl than in ci?
![image](https://github.com/nushell/nushell/assets/343840/62f152c7-41ba-44d2-8166-1231c5364522)

Reply:
I think in the CI the problem is that `nu!` is already running `nu` with come CLI arguments which doesn't seem to play nice with testbins. Since we're already running the system echo, I'd keep it and we can reconsider if it starts making problems.